packages/shared/src/getToken.ts

```diff
@@ -0,0 +1,160 @@
+import { inBrowser } from './browser';
+import type { ClerkStatus, GetTokenOptions, LoadedClerk } from './types';
+
+const POLL_INTERVAL_MS = 50;
+const MAX_POLL_RETRIES = 100; // 5 seconds of polling
+const TIMEOUT_MS = 10000; // 10 second absolute timeout
+
+type WindowClerk = LoadedClerk & {
+  status?: ClerkStatus;
+  loaded?: boolean;
+  on?: (event: 'status', handler: (status: ClerkStatus) => void, opts?: { notify?: boolean }) => void;
+  off?: (event: 'status', handler: (status: ClerkStatus) => void) => void;
+};
```

Comment:
[nit] Are all of these properties missing from `LoadedClerk`? If they are public we should consider updating the type if so